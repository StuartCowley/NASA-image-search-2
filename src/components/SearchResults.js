import propTypes from "prop-types";
import "../styles/SearchResults.css";

const SearchResults = ({ results }) => {
  if (!results.length) {
    return <p>Enter a valid search term</p>;
  } else {
    return (
      <div className="search-results">
        {results.map((result, index) => {
          return (
            <img
              src={result}
              alt="space"
              className="search-results__image"
              key={index}
            />
          );
        })}
      </div>
    );
  }
};

SearchResults.propTypes = {
  results: propTypes.arrayOf(propTypes.string),
};

export default SearchResults;
