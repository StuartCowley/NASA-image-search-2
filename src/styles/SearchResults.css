.search-results {
  margin-top: 30px;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-evenly;
  align-items: flex-start;
}

.search-results__image {
  width: 80%;
  border: 3px solid red;
  margin-top: 5px;
}

@media screen and (min-width: 768px) {
  .search-results__image {
    margin: 5px;
    height: auto;
  }
}

@media screen and (min-width: 1024px) {
  .search-results__image {
    margin: 5px;
    height: auto;
    width: 25%;
  }
  .search-results {
    margin-top: 50px;
  }
}
