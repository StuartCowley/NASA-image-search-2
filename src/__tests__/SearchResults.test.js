import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("SearchResults", () => {
  const validProps = ["test", "test"];

  test("component renders correctly", () => {
    const { asFragment } = render(<SearchResults results={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("renders placeholder text if no results", () => {
    const validProps = [];
    render(<SearchResults results={validProps} />);
    const placeholderText = screen.getByText("Enter a valid search term!");

    expect(placeholderText).toBeInTheDocument();
  });

  test("correct amount of images render on page if results exist", () => {
    render(<SearchResults results={validProps} />);
    const images = screen.getAllByAltText("space");

    expect(images.length).toEqual(2);
    expect(images[0]).toBeInTheDocument();
    expect(images[0]).toHaveClass("search-results__image");
  });
});
