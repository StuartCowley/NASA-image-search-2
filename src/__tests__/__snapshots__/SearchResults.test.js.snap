// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-results"
  >
    <img
      alt="space"
      class="search-results__image"
      src="test"
    />
    <img
      alt="space"
      class="search-results__image"
      src="test"
    />
  </div>
</DocumentFragment>
`;
